src/screens/Messages/Conversation/MessagesList.tsx

```diff
@@ -289,15 +313,110 @@ export function MessagesList({
           onScrollToIndexFailed={onScrollToIndexFailed}
           scrollEventThrottle={100}
           ListHeaderComponent={
-            <MaybeLoader isLoading={convo.isFetchingHistory} />
+            <MaybeLoader isLoading={convoState.isFetchingHistory} />
           }
         />
       </ScrollProvider>
-      <MessageInput
-        onSendMessage={onSendMessage}
-        scrollToEnd={scrollToEndNow}
-      />
+      {!isBlocking && !isBlockedBy ? (
+        <MessageInput
+          onSendMessage={onSendMessage}
+          scrollToEnd={scrollToEndNow}
+        />
+      ) : (
+        <BlockedFooter
+          recipient={recipient}
+          convoId={convoState.convo.id}
+          hasMessages={convoState.items.length > 0}
+          isBlocking={isBlocking}
+          isBlockedBy={isBlockedBy}
+        />
+      )}
       {showNewMessagesPill && <NewMessagesPill />}
     </Animated.View>
   )
 }
+
+function BlockedFooter({
+  recipient,
+  convoId,
+  hasMessages,
+  isBlocking,
+  isBlockedBy,
+}: {
+  recipient: Shadow<AppBskyActorDefs.ProfileViewBasic>
+  convoId: string
+  hasMessages: boolean
+  isBlocking: boolean
+  isBlockedBy: boolean
+}) {
+  const t = useTheme()
+  const {_} = useLingui()
+  const [__, queueUnblock] = useProfileBlockMutationQueue(recipient)
+
+  const leaveConvoControl = useDialogControl()
+  const reportDialogControl = useDialogControl()
+
+  if (!isBlocking && !isBlockedBy) return null
+
+  return (
+    <View style={[hasMessages && a.py_md, a.gap_lg]}>
```

Comment:
This could use just a little more padding on the bottom, little tight. Can be a size more on bottom than top if needed.